nuxeo-importer-stream/src/main/java/org/nuxeo/importer/stream/consumer/RedisDocumentMessageConsumer.java

```diff
@@ -0,0 +1,163 @@
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Consumes DocumentMessage and send them to Redis which can be used as Gatling feeder.
+ *
+ * @since 10.2
+ */
+public class RedisDocumentMessageConsumer extends AbstractConsumer<DocumentMessage> {
+    private static final Log log = LogFactory.getLog(RedisDocumentMessageConsumer.class);
+
+    protected static final String DEFAULT_REDIS_PREFIX = "imp";
+
+    protected static final String DOC_KEY_SUFFIX = "doc";
+
+    protected static final String DATA_KEY_SUFFIX = "data";
+
+    protected static final String FOLDER_KEY_SUFFIX = "folder";
+
+    protected static final String BATCH_ID_TAG = "_BATCH_ID_";
+
+    protected final byte[] addDocumentSHA;
+
+    protected final RedisExecutor redisExecutor;
+
+    protected final String redisPrefix;
+
+    public RedisDocumentMessageConsumer(String consumerId, String redisPrefix) {
+        super(consumerId);
+        if (StringUtils.isBlank(redisPrefix)) {
+            this.redisPrefix = DEFAULT_REDIS_PREFIX;
+        } else {
+            this.redisPrefix = redisPrefix;
+        }
+        RedisAdmin redisAdmin = Framework.getService(RedisAdmin.class);
+        try {
+            addDocumentSHA = redisAdmin.load("org.nuxeo.importer.stream", "add-document").getBytes("UTF-8");
+        } catch (IOException e) {
+            throw new NuxeoException("Cannot load Redis script", e);
+        }
+        redisExecutor = Framework.getService(RedisExecutor.class);
+    }
+
+    @Override
+    public void accept(DocumentMessage message) {
+        BlobInfo blobInfo = message.getBlobInfo();
+        String blobPath = "";
+        String blobFilename = "";
+        String blobMimeType = "";
+        String properties;
+        try {
+            if (blobInfo == null || StringUtils.isBlank(blobInfo.filename)) {
+                properties = getProperties(message);
+            } else {
+                blobPath = blobInfo.filename;
+                blobMimeType = blobInfo.mimeType;
+                blobFilename = Paths.get(blobInfo.filename).getFileName().toString();
+                properties = getPropertiesWithBlob(message);
+            }
+            String parentPath = message.getParentPath();
+            if (parentPath.startsWith("/")) {
+                parentPath = parentPath.substring(1);
+            }
+            String key = Paths.get(parentPath, message.getName()).toString();
+            String docKey = redisPrefix + ":" + DOC_KEY_SUFFIX;
+            String folderKey = redisPrefix + ":" + FOLDER_KEY_SUFFIX;
+            String dataKey = redisPrefix + ":" + DATA_KEY_SUFFIX + ":" + key;
+            String level = String.valueOf(key.split("/").length);
+            String url = URLEncoder.encode(key, "UTF-8").replaceAll("%2F", "/");
```

Comment:
`URLEncoder` is probably the wrong class, it's designed to encode query parameters, not paths. It all depends on what the decoder expects, and it's not clear to me where the decoder is.
(Our `URIUtils.quoteURIPathComponent` may be a better choice if that's what the decoder expects.)

Reply:
This URL will be used in Gatling simulation, it matches Nuxeo URL for the document, so path separators are preserved while non-ASCII char are encoded.

Reply:
Ok so if it's for an encoded path part of Nuxeo URL then `URLEncoder` is wrong. It will encode spaces as `+` instead of `%20`. You should really use `URIUtils.quoteURIPathComponent` instead.